Always maintain the invariant for TransactionSignature

sorpaas opened the pull request:
The `TransactionSignature` type was designed to always maintain the invariant for EIP-2, but this wasn't later properly done for EIP-1559, EIP-2930, etc.

sorpaas committed:
Always use TransactionSignature wrap

sorpaas committed:
Always maintain the invariant for TransactionSignature

sorpaas committed:
use -> pub use

RomarQ commented on src/transaction/mod.rs:
```diff
@@ -428,9 +430,12 @@ mod tests {
-			odd_y_parity: false,
-			r: hex!("36b241b061a36a32ab7fe86c7aa9eb592dd59018cd0443adc0903590c16b02b0").into(),
-			s: hex!("5edcc541b4741c5cc6dd347c5ed9577ef293a62787b4510465fadbfe39ee4094").into(),
+			signature: eip2930::TransactionSignature::new(
```
```suggestion
			signature: eip1559::TransactionSignature::new(
```

RomarQ replied:
Nevermind, just saw that it reuses eip2930::TransactionSignature.

RomarQ replied:
Maybe it could be in a common module that shares the common types like `TransactionSignature`, `AccessList` and `TransactionAction`, just to avoid confusion?

sorpaas replied:
The idea is to reuse unchanged struct from older hard forks to newer hard forks.

* We have legacy -> EIP-2930 -> EIP-1559 -> EIP-7702.
* In legacy, a `TransactionSignature` is defined. This becomes `legacy::TransactionSignature`.
* In EIP-2930, the format of `TransactionSignature` is changed, so we define a `eip2930::TransactionSignature`.
* The format of signature is unchanged in EIP-1559 and EIP-7702, so in `eip1559` and `eip-7702` mods, we `pub use super::eip2930::TransactionSignature`.

RomarQ commented on src/transaction/mod.rs:
```diff
@@ -463,13 +468,19 @@ mod tests {
-			odd_y_parity: false,
-			r: hex!("36b241b061a36a32ab7fe86c7aa9eb592dd59018cd0443adc0903590c16b02b0").into(),
-			s: hex!("5edcc541b4741c5cc6dd347c5ed9577ef293a62787b4510465fadbfe39ee4094").into(),
+			signature: eip2930::TransactionSignature::new(
```
Nevermind, just saw that it reuses eip2930::TransactionSignature.

RomarQ reviewed (approved):
LGTM

Just added a small remark

sorpaas committed:
Make authorization list item signature malleable

When the signature is invalid, the specification still asks that only
the item is ignored, but not that decoding fails.
